config/v1/types_infrastructure.go

```diff
@@ -600,9 +600,300 @@ type OvirtPlatformStatus struct {
+// computeCluster, networks, datastore and resourcePool - to provision virtual machines.
+type VSpherePlatformTopology struct {
+	// datacenter is the vCenter datacenter in which virtual machines will be located
+	// and defined as the failure domain.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	Datacenter string `json:"datacenter"`
+
+	// computeCluster as the failure domain
+	// This is required to be a path
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	ComputeCluster string `json:"computeCluster,omitempty"`
+
+	// Hosts has information required for placement of machines on VSphere hosts.
+	// +optional
+	Hosts VSpherePlatformFailureDomainHosts `json:"hosts,omitempty"`
+
+	// networks is the list of port group network names within this failure domain.
+	// Currently, we only support a single interface per RHCOS virtual machine.
+	// The available networks (port groups) can be listed using
+	// govc ls 'network/*'
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxItems=1
+	// +kubebuilder:validation:MinItems=1
+	Networks []string `json:"networks,omitempty"`
+
+	// datastore is the name or inventory path of the datastore in which the
+	// virtual machine is created/located.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=2048
+	Datastore string `json:"datastore,omitempty"`
+}
+
+type VSpherePlatformFailureDomainHosts struct {
+	// vmGroupName is the Virtual Machine Group name configured
+	// within a vCenter cluster that is associated with
+	// the corresponding Host Group.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	VMGroupName string `json:"vmGroupName"`
+
+	// hostGroupName is the Host Group name configured
+	// within a vCenter cluster defining a group
+	// of ESXi hosts.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=80
+	HostGroupName string `json:"hostGroupName"`
+}
+
+// VSpherePlatformVCenterSpec stores the vCenter connection fields.
+// This is used by the vSphere CCM.
+type VSpherePlatformVCenterSpec struct {
+
+	// server is the fully-qualified domain name or the IP address of the vCenter server.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MaxLength=255
+	Server string `json:"server"`
+
+	// port is the TCP port that will be used to communicate to
+	// the vCenter endpoint.
+	// When omitted, this means the user has no opinion and
+	// it is up to the platform to choose a sensible default,
+	// which is subject to change over time.
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=32767
+	// +optional
+	Port int `json:"port,omitempty"`
```

Comment:
Don't use a regular `int` as it varies depending on the Machine, `int32` should suffice here